appservice/src/main/java/nu/yona/server/subscriptions/rest/UserController.java

```diff
@@ -119,6 +135,24 @@ public static Link getUserLink(UUID userID, boolean includePrivateData) {
 		public UserResource(UserDTO user) {
 			super(user);
 		}
+		
+		@JsonProperty("_embedded")
+		public Map<String, List<BuddyController.BuddyResource>> getEmbeddedResources() {
+			if(getContent().getPrivateData() != null)
+			{
+				Optional<Set<BuddyDTO>> buddies = getContent().getPrivateData().getBuddies();
+				if(buddies.isPresent()) {
+					return Collections.singletonMap(UserDTO.BUDDIES_REL_NAME,
+						new BuddyController.BuddyResourceAssembler(getContent().getID()).toResources(buddies.get())); 
+				}					
+			}
+			return Collections.emptyMap();
+		}
```

Comment:
Remove the nesting here by moving the return of the empty collection up into the first if statement.